zkstack_cli/crates/zkstack/src/commands/chain/notify_server_calldata.rs

```diff
@@ -0,0 +1,125 @@
+use std::path::Path;
+
+use anyhow::Context;
+use clap::Parser;
+use serde::{Deserialize, Serialize};
+use xshell::Shell;
+use zkstack_cli_common::{forge::ForgeScriptArgs, logger};
+use zksync_types::Address;
+
+use super::{
+    gateway_common::{
+        get_gateway_migration_state, GatewayMigrationProgressState, MigrationDirection,
+    },
+    utils::{display_admin_script_output, get_default_foundry_path},
+};
+use crate::{
+    accept_ownership::{
+        notify_server_migration_from_gateway, notify_server_migration_to_gateway, AdminScriptMode,
+        AdminScriptOutput,
+    },
+    messages::message_for_gateway_migration_progress_state,
+};
+
+#[derive(Debug, Serialize, Deserialize, Parser)]
+pub struct NotifyServerCallsArgs {
+    pub l1_bridgehub_addr: Address,
+    pub l2_chain_id: u64,
+    pub l1_rpc_url: String,
+}
+
+pub async fn get_notify_server_calls(
+    shell: &Shell,
+    forge_args: &ForgeScriptArgs,
+    forge_path: &Path,
+    args: NotifyServerCallsArgs,
+    direction: MigrationDirection,
+) -> anyhow::Result<AdminScriptOutput> {
+    let admin_call_output = match direction {
+        MigrationDirection::FromGateway => {
+            notify_server_migration_from_gateway(
+                shell,
+                forge_args,
+                forge_path,
+                AdminScriptMode::OnlySave,
+                args.l2_chain_id,
+                args.l1_bridgehub_addr,
+                args.l1_rpc_url,
+            )
+            .await
+        }
+        MigrationDirection::ToGateway => {
+            notify_server_migration_to_gateway(
+                shell,
+                forge_args,
+                forge_path,
+                AdminScriptMode::OnlySave,
+                args.l2_chain_id,
+                args.l1_bridgehub_addr,
+                args.l1_rpc_url,
+            )
+            .await
+        }
+    }?;
+
+    Ok(admin_call_output)
+}
+
+#[derive(Debug, Serialize, Deserialize, Parser)]
+pub struct NotifyServerCalldataArgs {
+    #[clap(flatten)]
+    pub params: NotifyServerCallsArgs,
+    pub l2_rpc_url: Option<String>,
+    pub gw_rpc_url: Option<String>,
+    pub no_cross_check: Option<bool>,
```

Comment:
please use `default_missing_value` param for bools. it'd allow to use --no-cross-check and no  --no-cross-check=true

also, probably we want to have it non option and use false as a default